addons/mrp/models/stock_move.py

```diff
@@ -345,6 +345,8 @@ def create(self, vals_list):
                 values['origin'] = mo._get_origin()
                 values['group_id'] = mo.procurement_group_id.id
                 values['propagate_cancel'] = mo.propagate_cancel
+                if mo.qty_producing > 0:
```

Comment:
```suggestion
                if mo.state in ['progress', 'to_close']:
```
Move that condition inside:
```py
if values.get('raw_material_production_id', False):
```
As it only matters for components, not byproducts.

Reply:
Hello @clesgow,

I think we need to go with the quantity condition because the progress state is active even when just one component is consumed, but the product is not produced, and the specification is not fulfilled. Additionally, the barcode functionality of the split move in production, when we consume only some of the required components, will also break.

If you still want the state condition, we can add it along with the quantity condition for a double check.

Thank You.